packages/webpack-cli/lib/groups/resolveConfig.js

```diff
@@ -166,6 +168,7 @@ const finalize = async (moduleObj, args) => {
 
             // `Promise` may return `Function`
             if (typeof rawConfig === 'function') {
+                isFunctionalConfig = true;
                 // when config is a function, pass the env from args to the config function
                 rawConfig = await rawConfig(env, args);
```

Comment:
I think we need to map again if this returns an array.

Reply:
Do you mean when we return multiple configurations?

Reply:
Yes because when we have `module.exports = () => [config, config]` here it becomes `rawconfig = [ config, config ]` with which the next return statement becomes `[ [ config, config ] ]`.

Reply:
Let's refactor it, just put configuration in array

Reply:
Can you clarify?

Reply:
```js
const result = [];

await Promise.all(() => { /* async code here */ }).map(item => result.push(item))
```